src/components/Counter.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  decrement,
+  increment,
+  incrementByAmount,
+  reset,
+} from '../redux/slices/counter';
+
+export default function Counter() {
+  const count = useSelector((state) => state.counter.value);
+  const dispatch = useDispatch();
+  const [value, setValue] = useState(0);
+
+  const handleIncrement = () => {
+    dispatch(increment());
+  };
+  const handleDecrement = () => {
+    dispatch(decrement());
+  };
+  const handleReset = () => {
+    dispatch(reset());
+  };
+  const handleIncrementByValue = (e) => {
+    e.preventDefault();
+    dispatch(incrementByAmount(value));
+  };
```

Comment:
create a custom hook that will allow us to consume those actions and states then we can call that hook after destructing the function or action we need [reference](https://redux-toolkit.js.org/api/other-exports/#bindactioncreators)

Reply:
that hook can contain two things reduces, and dispatched actions but they are different hook but in the same file